### PR TITLE
replaced "divisor" with "delimiter" for methods split,rsplit and split_floats.

### LINKED ISSUE #29429: String.rsplit and String.split are the same
Maybe i am just misunderstanding but this is what i expected and what i'v got instead:
var updates : String = "path/morePath/file.exe"
Both print(updates.rsplit("/")) and print(updates.split("/")) returns ["path","morePath","file.exe"] while i would expect print(updates.rsplit("/")) to return ["file.exe","morePath","path"]. Even in the documentation it states that the result is the same. If so, then whats the difference between them?

> PoolStringArray split ( String divisor, bool allow_empty=True, int maxsplit=0 )
> Splits the string by a divisor string and returns an array of the substrings.
> 
> Example: "One,Two,Three" will return ["One","Two","Three"] if split by ",".
> 
> If maxsplit is given, at most maxsplit number of splits occur, and the remainder of the string is returned as the final element of the list (thus, the list will have at most maxsplit+1 elements)

> PoolStringArray rsplit ( String divisor, bool allow_empty=True, int maxsplit=0 )
> Splits the string by a divisor string and returns an array of the substrings, starting from right.
> 
> Example: "One,Two,Three" will return ["One","Two","Three"] if split by ",".
> 
> If maxsplit is specified, then it is number of splits to do, default is 0 which splits all the items.

### PR DESCRIPTION
**Pull request by zakstam**
Supersedes #29430.
Fixes #29429.

### CODE REVIEW

**Comment by aaronfranke** [comment]
This change should be placed along with your other changes in your previous PR.

Please see [this article](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html) for how to rebase, squash, and force push so that you just have one commit. I also created [this video](https://www.youtube.com/watch?v=0XKy8YpP5Ww) to show how it's done with a GUI client.

**Comment by zaksnet** [comment]
> This change should be placed along with your other changes in your previous PR.
> Please see this article for how to rebase, squash, and force push so that you just have one commit. I also created this video to show how it's done with a GUI client.

Thank you Aaron, great video!

**Comment by aaronfranke** [comment]
git rebase -i HEAD~7

Change all of your commits from "pick" to "fixup", except for the top one. Then:

    git push --force

**Comment by zaksnet** [comment]
> ```
> git rebase -i HEAD~7
> ```
> 
> Change all of your commits from "pick" to "fixup", except for the top one. Then:
> 
> ```
> git push --force
> ```

I probably did something very wrong, i see commits from other users also:
Screenshot_8

**Comment by aaronfranke** [comment]
@zaksnet It doesn't look like your branch is checked out, since I *only* see other people's commits. If you wish to try a graphical client, this is easy to do in GitKraken:



When you manage to do it correctly, on the top where it says "Commits" the number will be "1".

**Comment by zaksnet** [comment]
> @zaksnet It doesn't look like your branch is checked out, since I _only_ see other people's commits. If you wish to try a graphical client, this is easy to do in GitKraken:
> 
> 
> 
> When you manage to do it correctly, on the top where it says "Commits" the number will be "1".

I think i found gitkrakken harder to use than cmd. It seems to be fine now, i removed the commits from other people and added fixup to my own commits. Anyway, i think i'm done with git for now, thanks for the help.

**Comment by aaronfranke** [comment]
Awesome job, the git history looks good now. But it looks like you missed a few spots. The argument on line 665 is still "divisor" and the docs for `split` and `split_floats` on lines 710 to 736 need updating.

**Comment by zaksnet** [comment]
> Awesome job, the git history looks good now. But it looks like you missed a few spots. The argument on line 665 is still "divisor" and the docs for `split` and `split_floats` on lines 710 to 736 need updating.

Thank you!

**Review by aaronfranke — APPROVED** [review]
The documentation changes and git history all look good now, and are good improvements.

Closes https://github.com/godotengine/godot/pull/29430, closes https://github.com/godotengine/godot/issues/29429, @akien-mga, @godotengine/documentation (this doesn't work?)

File: doc/classes/String.xml [hunk 1]
```diff
@@ -662,16 +662,17 @@
...
 			<argument index="2" name="maxsplit" type="int" default="0">
 			</argument>
 			<description>
-				Splits the string by a [code]divisor[/code] string and returns an array of the substrings, starting from right.
-				[b]Example:[/b] [code]"One,Two,Three"[/code] will return [code]["One","Two","Three"][/code] if split by [code]","[/code].
-				If [code]maxsplit[/code] is specified, then it is number of splits to do, default is 0 which splits all the items.
+				Splits the string by a [code]delimiter[/code] string and returns an array of the substrings, starting from right.
+				The splits in the returned array are sorted in the same order as the original string, from left to right.
+				If [code]maxsplit[/code] is specified, it defines the number of splits to do from the right up to [code]maxsplit[/code]. The default value of 0 means that all items are split, thus giving the same result as [method split].
+				[b]Example:[/b] [code]"One,Two,Three,Four"[/code] will return [code]["Three","Four"][/code] if split by [code]","[/code] with maxsplit of [code]"2"[/code].
```
Comment (akien-mga) [inline doc/classes/String.xml]: As mentioned previously, the integer value `2` should not be within double quotes, which are the text representation of Strings. So it should be `with a [code]maxsplit[/code] value of 2`.

It would also be great if you could do similar changes to the `split` description for consistency.

Otherwise the changes are great, thanks a lot for putting up with the cumbersome Git workflow. You'll see that it gets easier :)

Comment (zaksnet) [reply doc/classes/String.xml]: > As mentioned previously, the integer value 2 should not be within double quotes, which are the text representation of Strings. So it should be with a [code]maxsplit[/code] value of 2.

So values should not be within [code][/code] tags? Should i also remove the tags from [code]","[/code] since its a value for the delimiter?

> It would also be great if you could do similar changes to the split description for consistency.

Sure!

> Otherwise the changes are great, thanks a lot for putting up with the cumbersome Git workflow. You'll see that it gets easier :)

Thank you! I think i'm getting the hang of it now that i switched to cmd. Managed to remove the commits from master (#29430) and close it since it was a duplicate to this one. I tend to make a mess when i learn and thats mostly ok if i'm by myself but now i felt i'm messing with other peoples work also.

Comment (aaronfranke) [reply doc/classes/String.xml]: Don't worry about messing up other people's work. The absolute worst case scenario is you mess up your copy of other people's work. But it seems that you have it figured out now.

Comment (akien-mga) [reply doc/classes/String.xml]: > So values should not be within [code][/code] tags? Should i also remove the tags from [code]","[/code] since its a value for the delimiter? Or just remove quotes from the integer value?

I don't have a strong opinion on that one, I guess you can use either `2` or `[code]2[/code]`. To me the latter emphasizes that it's the programming value "2" which represents an integer, while the former would be the "usual" number 2 that we refer to in plain English. (Don't know if that makes sense...)

**Comment by akien-mga** [comment]
Great work, thanks a lot! :)

**Comment by akien-mga** [comment]
Cherry-picked for 3.1.2.
